plugins/kubernetes-backend/src/service/KubernetesBuilder.ts

```diff
@@ -43,6 +43,16 @@ export interface KubernetesEnvironment {
   config: Config;
 }
 
+export type KubernetesBuilderReturn = Promise<{
+  router: express.Router;
+  clusterDetails?: ClusterDetails[];
+  clusterSupplier?: KubernetesClustersSupplier;
+  customResources?: CustomResource[];
+  fetcher?: KubernetesFetcher;
+  objectsProvider?: KubernetesObjectsProvider;
+  serviceLocator?: KubernetesServiceLocator;
```

Comment:
These are always present in the happy path right? I think it's likely a bit nicer to make them required and then cast the early return with `as KubernetesBuilderReturn`

Reply:
Sounds good. Fixed in 81248c2d1fb6fd6cb5ee8fbe901b3312b00730e6!